addons/l10n_id_efaktur/tests/test_l10n_id_efaktur.py

```diff
@@ -0,0 +1,91 @@
+        """
+        1) contact with l10n_id_pkp=True, l10n_id_kode_transaksi="01"
+        2) tax: amount=10, type_tax_use=sale, price_include=True
+        3) invoice with partner_id=contact, journal=customer invoices,
+        """
+        super().setUp()
+
+        self.maxDiff = 1500
+        # change company info for csv detai later
+        self.env.company.country_id = self.env.ref('base.id')
+        self.env.company.street = "test"
+        self.env.company.phone = "12345"
+
+        self.partner_id = self.env['res.partner'].create({"name": "l10ntest", "l10n_id_pkp": True, "l10n_id_kode_transaksi": "01", "l10n_id_nik": "12345"})
+        self.tax_id = self.env['account.tax'].create({"name": "test tax", "type_tax_use": "sale", "amount": 10.0, "price_include": True})
+
+        self.efaktur = self.env['l10n_id_efaktur.efaktur.range'].create({'min': '0000000000001', 'max': '0000000000010'})
+        self.out_invoice_1 = self.env['account.move'].create({
+            'type': 'out_invoice',
+            'partner_id': self.partner_id.id,
+            'invoice_date': '2019-05-01',
+            'date': '2019-05-01',
+            'invoice_line_ids': [
+                (0, 0, {'name': 'line1', 'price_unit': 110.0, 'tax_ids': self.tax_id.ids}),
+            ],
+            'l10n_id_kode_transaksi': "01",
+        })
+        self.out_invoice_1.post()
+
+        self.out_invoice_2 = self.env['account.move'].create({
+            'type': 'out_invoice',
+            'partner_id': self.partner_id.id,
+            'invoice_date': '2019-05-01',
+            'date': '2019-05-01',
+            'invoice_line_ids': [
+                (0, 0, {'name': 'line1', 'price_unit': 110.11, 'quantity': 400, 'tax_ids': self.tax_id.ids})
+            ],
+            'l10n_id_kode_transaksi': '01'
+        })
+        self.out_invoice_2.post()
+
+    def test_efaktur_csv_output_1(self):
+        """
+        Test to ensure that the output csv data contains tax-excluded prices regardless of whether the tax configuration is tax-included or tax-excluded.
+        Current test is using price of 110 which is tax-included with tax of amount 10%. So the unit price listed has to be 100 whereas the original result would have 110 instead.
+        """
+        # to check the diff when test fails
+
+        efaktur_csv_output = self.out_invoice_1._generate_efaktur_invoice(',')
+        output_head = '%s%s%s' % (
+            _csv_row(FK_HEAD_LIST, ','),
+            _csv_row(LT_HEAD_LIST, ','),
+            _csv_row(OF_HEAD_LIST, ','),
+        )
+        # remaining lines
+        line_4 = '"FK","01","0","0000000000001","5","2019","1/5/2019","12345","l10ntest","","100","10","0","","0","110","0","0","INV/2019/0001 12345","0"\n'
+        line_5 = '"FAPR","000000000000000","YourCompany","test","","","","","","","","","","12345"\n'
+        line_6 = '"OF","","","100","1.0","100","0","100","10","0","0"\n'
+
+        efaktur_csv_expected = output_head + line_4 + line_5 + line_6
```

Comment:
Don't limit yourself to numbered variables (`line_4`) and hardcoded patterns (`%s%s%s`)
```suggestion
        efaktur_csv_expected = ''.join([
            _csv_row(FK_HEAD_LIST, ','),
            _csv_row(LT_HEAD_LIST, ','),
            _csv_row(OF_HEAD_LIST, ','),
            _csv_row(["FK","01","0","0000000000001","5","2019","1/5/2019","12345","l10ntest","","100","10","0","","0","110","0","0","INV/2019/0001 12345","0"], ','),
            _csv_row(["FAPR","000000000000000","YourCompany","test","","","","","","","","","","12345"], ','),
            _csv_row(["OF","","","100","1.0","100","0","100","10","0","0"], ','),
        ])
```
or
```suggestion
        efaktur_csv_expected = (
            '"FK","KD_JENIS_TRANSAKSI","FG_PENGGANTI","NOMOR_FAKTUR","MASA_PAJAK","TAHUN_PAJAK","TANGGAL_FAKTUR","NPWP","NAMA","ALAMAT_LENGKAP","JUMLAH_DPP","JUMLAH_PPN","JUMLAH_PPNBM","ID_KETERANGAN_TAMBAHAN","FG_UANG_MUKA","UANG_MUKA_DPP","UANG_MUKA_PPN","UANG_MUKA_PPNBM","REFERENSI","KODE_DOKUMEN_PENDUKUNG"\n'
            '"LT","NPWP","NAMA","JALAN","BLOK","NOMOR","RT","RW","KECAMATAN","KELURAHAN","KABUPATEN","PROPINSI","KODE_POS","NOMOR_TELEPON"\n'
            '"OF","KODE_OBJEK","NAMA","HARGA_SATUAN","JUMLAH_BARANG","HARGA_TOTAL","DISKON","DPP","PPN","TARIF_PPNBM","PPNBM"\n'
            '"FK","01","0","0000000000001","5","2019","1/5/2019","12345","l10ntest","","100","10","0","","0","110","0","0","INV/2019/0001 12345","0"\n'
            '"FAPR","000000000000000","YourCompany","test","","","","","","","","","","12345"\n'
            '"OF","","","100","1.0","100","0","100","10","0","0"\n'
        )
```
(first one is a bit more readable but we are not testing `_csv_row` and `*_HEAD_LIST`)